spras/btb.py

```diff
@@ -0,0 +1,188 @@
+# need to define a new btb class and contain the following functions
+# - generate_inputs
+# - run
+# - parse_output
+
+import warnings
+from pathlib import Path
+
+import pandas as pd
+
+from spras.containers import prepare_volume, run_container
+# from spras.interactome import (
+#     convert_undirected_to_directed,
+#     reinsert_direction_col_directed,
+# )
+# what type of directionality does btb support?
+
+from spras.prm import PRM
+
+__all__ = ['BowtieBuilder']
+
+class BowtieBuilder(PRM):
+    required_inputs = ['sources', 'targets', 'edges']
+
+    #generate input taken from meo.py beacuse they have same input requirements
+    @staticmethod
+    def generate_inputs(data, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in BowtieBuilder.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing")
+        print("FILEMAP NAME: ", filename_map)
+        print("DATA HEAD: ")
+        print( data.node_table.head())
+        print("DATA INTERACTOME: ")
+        print(data.interactome.head())
+
+        # Get sources and write to file, repeat for targets
+        # Does not check whether a node is a source and a target
+        for node_type in ['sources', 'targets']:
+            nodes = data.request_node_columns([node_type])
+            if nodes is None:
+                raise ValueError(f'No {node_type} found in the node files')
+
+            # TODO test whether this selection is needed, what values could the column contain that we would want to
+            # include or exclude?
+            nodes = nodes.loc[nodes[node_type]]
+            if(node_type == "sources"):
+                nodes.to_csv(filename_map["sources"], sep= '\t', index=False, columns=['NODEID'], header=False)
+                print("NODES: ")
+                print(nodes)
+            elif(node_type == "targets"):
+                nodes.to_csv(filename_map["targets"], sep= '\t', index=False, columns=['NODEID'], header=False)
+                print("NODES: ")
+                print(nodes)
+
+
+        # Create network file
+        edges = data.get_interactome()
+
+        # Format network file
+        #unsure if formating network file is needed
```

Comment:
I don't think this is needed - this is for formatting the output edge files in the standard format, right?

Reply:
removed